### ISSUE TITLE
Has GPG key of apt repo expired?

### ISSUE THREAD

**Issue by MatthewFagan** [kubernetes/release#2860]
apt-get update returns the following error:
Err:4 https://packages.cloud.google.com/apt kubernetes-xenial InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05

This occurred previously in https://github.com/kubernetes/release/issues/1982. Is there a problem with new key distribution? Is this a known issue and expected to get fixed soon?

**Comment by dims** [comment]
xref : https://github.com/kubernetes/release/issues/2862

**Comment by lukeweller** [comment]
I'm seeing the same issue on my Ubuntu 20.04.5 machine:
```
$ sudo apt update             
...
Err:2 https://packages.cloud.google.com/apt kubernetes-xenial InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
...
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://packages.cloud.google.com/apt kubernetes-xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
W: Failed to fetch https://apt.kubernetes.io/dists/kubernetes-xenial/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by delta1** [comment]
This fixed it for me 

```bash
curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
```

**Comment by bueti** [comment]
it seems like the new key isn't deployed everywhere because even when adding the new key manually I get intermittent errors

**Comment by afolarin** [comment]
> it seems like the new key isn't deployed everywhere because even when adding the new key manually I get intermittent errors

Yes, same here.

**Comment by elitistphoenix** [comment]
Ditto as per @afolarin 
Hopefully nothing dodgy going on.

**Comment by sftim** [comment]
https://github.com/kubernetes/enhancements/issues/1731 is tracking work to address this

**Comment by raykrueger** [comment]
The apt-key-gpg url seems to be giving 500 errors from random places in the world. I setup a health check with AWS Route 53 and see the following results. The dns and https GETs will happen from each of the regions below.  Hopefully this is useful data.

Asia Pacific (Tokyo) | 15.177.42.9 | Feb 26, 2023 7:30:48 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.199.110 |  
-- | -- | -- | -- | --
Asia Pacific (Tokyo) | 15.177.46.9 | Feb 26, 2023 7:30:48 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.207.46 |  
Asia Pacific (Singapore) | 15.177.54.9 | Feb 26, 2023 7:30:48 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.12.139 |  
Asia Pacific (Singapore) | 15.177.50.11 | Feb 26, 2023 7:30:48 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 172.217.194.102 |  
Asia Pacific (Sydney) | 15.177.58.9 | Feb 26, 2023 7:30:50 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 142.250.71.78 |  
Asia Pacific (Sydney) | 15.177.62.9 | Feb 26, 2023 7:30:50 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 142.250.71.78 |  
EU (Ireland) | 15.177.34.9 | Feb 26, 2023 7:30:48 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 74.125.193.101 |  
EU (Ireland) | 15.177.38.9 | Feb 26, 2023 7:30:47 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 74.125.193.139 |  
South America (São Paulo) | 15.177.26.9 | Feb 26, 2023 7:30:48 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 142.250.218.174 |  
South America (São Paulo) | 15.177.30.9 | Feb 26, 2023 7:30:48 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.79.206 |  
US East (N. Virginia) | 15.177.6.9 | Feb 26, 2023 7:30:46 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 172.253.62.113 |  
US East (N. Virginia) | 15.177.2.11 | Feb 26, 2023 7:30:46 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 142.251.163.113 |  
US West (N. California) | 15.177.10.9 | Feb 26, 2023 7:30:47 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.46.238 |  
US West (N. California) | 15.177.14.9 | Feb 26, 2023 7:30:47 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.214.142 |  
US West (Oregon) | 15.177.22.9 | Feb 26, 2023 7:30:46 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 172.217.14.206 |  
US West (Oregon) | 15.177.18.9 | Feb 26, 2023 7:30:46 PM UTC | Failure: HTTP Status Code 500, Internal Server Error. Resolved IP: 142.251.33.110

**Comment by bueti** [comment]
once again, google rotates keys w/o updating all locations 🤦

**Comment by bueti** [comment]
workaround suggested by google support: https://github.com/GoogleCloudPlatform/gcsfuse/issues/961#issuecomment-1445711531

**Comment by dims** [comment]
See work around with new URL here: https://github.com/kubernetes/k8s.io/pull/4837#issuecomment-1446426585

**Comment by cpanato** [comment]
key are back  https://packages.cloud.google.com/apt/doc/apt-key.gpg

/close

**Comment by raykrueger** [comment]
Confirmed that the gpg health checks are all passing now.



Health checker region | Health checker IP | Last checked | Status |  
-- | -- | -- | -- | --
Asia Pacific (Tokyo) | 15.177.42.9 | Feb 27, 2023 7:38:31 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.199.110 |  
Asia Pacific (Tokyo) | 15.177.46.9 | Feb 27, 2023 7:38:49 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.42.174 |  
Asia Pacific (Singapore) | 15.177.54.9 | Feb 27, 2023 7:38:44 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.4.139 |  
Asia Pacific (Singapore) | 15.177.50.11 | Feb 27, 2023 7:38:54 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 172.253.118.102 |  
Asia Pacific (Sydney) | 15.177.58.9 | Feb 27, 2023 7:38:46 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.204.14 |  
Asia Pacific (Sydney) | 15.177.62.9 | Feb 27, 2023 7:38:44 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.66.206 |  
EU (Ireland) | 15.177.34.9 | Feb 27, 2023 7:38:28 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 74.125.193.113 |  
EU (Ireland) | 15.177.38.9 | Feb 27, 2023 7:38:50 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 74.125.193.100 |  
South America (São Paulo) | 15.177.26.9 | Feb 27, 2023 7:38:40 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.132.238 |  
South America (São Paulo) | 15.177.30.9 | Feb 27, 2023 7:38:34 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.129.174 |  
US East (N. Virginia) | 15.177.6.9 | Feb 27, 2023 7:38:55 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 172.253.122.101 |  
US East (N. Virginia) | 15.177.2.11 | Feb 27, 2023 7:38:45 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.16.138 |  
US West (N. California) | 15.177.10.9 | Feb 27, 2023 7:38:41 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.250.189.174 |  
US West (N. California) | 15.177.14.9 | Feb 27, 2023 7:38:34 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.46.238 |  
US West (Oregon) | 15.177.22.9 | Feb 27, 2023 7:38:39 PM UTC | Success: HTTP Status Code 200, OK. Resolved IP: 142.251.33.78
